File: Facility/Mixer.cpp
/**
 * @file Mixer.hpp
 * @author 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @brief file berisi definisi class Mixer
 * @version 0.1
 * @date 2019-03-20
 */

/* NIM/Nama  : 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * Nama file : Mixer.hpp
 * Topik     : Tugas Besar 1 IF2210 - Pemograman Berorientasi Objek
 * Tanggal   : 20 Maret 2019
 * Deskripsi : Header Mixer.cpp */

#include "Mixer.hpp"
#include "../SideProduct/FailedSideProduct.hpp"
#include <string.h>
using namespace std;

/**
 * @brief Construct a new Mixer object
 * 
 */
Mixer::Mixer(){
	setY(3);
	setX(10);
}

/**
 * @brief Destroy the Mixer object
 * 
 */
Mixer::~Mixer(){
	
}

/**
 * @brief Method to render the Facility type character to map
 * 
 * @return char 
 */
char Mixer::render(){
	return 'M';
}

File: Java/model/facility/Mixer.java
/**
 * Mixer.java
 * <p>
 * File berisi interface Mixer
 * </p>
 * @author 13515125/Muhammad Rafid Amrullah, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @version 2.0
 * @since 2019-04-21
 */

import java.io.*;
import java.lang.*;
import java.util.*;

/**
 * <p>
 * Class for rendering object to Mixer
 * </p>
 */
public class Mixer extends Facility
{
	/**
	 * Mixer constructor
	 * @param y
	 * @param x
	 * @return Character
	 */
	public Mixer(int y, int x){
		setY(y);
		setX(x);
	}

	/**
	 * Render Well to map
	 * @return Character
	 */
	public Character render(){
		return 'M';
	}
}
